autofaiss/indices/distributed.py

```diff
@@ -355,7 +378,315 @@ def add_embeddings_to_index(
+
+    # Read embeddings
+    with Timeit("-> Reading embeddings", indent=2):
+        embedding_reader = EmbeddingReader(
+            embedding_root_dir,
+            file_format="parquet",
+            embedding_column=embedding_column_name,
+            meta_columns=id_columns,
+        )
+    
+    # Compute memory available for adding embedding to index
+    metadata = IndexMetadata(index_key, embedding_reader.count, embedding_reader.dimension, make_direct_map)
+    index_size = metadata.estimated_index_size_in_bytes()
+    memory_available_for_adding = cast_bytes_to_memory_string(
+        cast_memory_to_bytes(current_memory_available) - index_size
+    )
+
+    write_ids_df_to_parquet_fn = \
+        get_write_ids_df_to_parquet_fn(
+            ids_root_dir=ids_dest_dir
+        ) if id_columns else None
+
+    optimize_index_fn = \
+        get_optimize_index_fn(
+            embedding_reader=embedding_reader,
+            index_key=index_key,
+            index_path=index_dest_path,
+            index_infos_path=index_infos_dest_path,
+            use_gpu=use_gpu,
+            save_on_disk=True,
+            max_index_query_time_ms=max_index_query_time_ms,
+            min_nearest_neighbors_to_retrieve=min_nearest_neighbors_to_retrieve
+        )
+
+    # Add embeddings to index
+    return add_embeddings_fn(
+        embedding_reader=embedding_reader,
+        memory_available_for_adding=memory_available_for_adding,
+        embedding_ids_df_handler=write_ids_df_to_parquet_fn,
+        index_optimizer=optimize_index_fn
+    )
+
+
+def _add_embeddings_to_index_local(
+    trained_index_path: str,
+    embedding_reader: EmbeddingReader,
+    memory_available_for_adding: int,
+    make_direct_map: bool,
+    embedding_ids_df_handler: Optional[Callable[[pd.DataFrame, int], Any]],
+    index_optimizer: Optional[Callable]
+) -> Dict[str, str]:
+    """Add embeddings to small index"""
+
+    with tempfile.TemporaryDirectory() as tmp:
+        trained_index_tmp_path = os.path.join(tmp, "index")
+        trained_index = _load_index(trained_index_path, trained_index_tmp_path)
+        
+        batch_size = get_optimal_batch_size(embedding_reader.dimension, memory_available_for_adding)
+        all_ids = []
+        for (vec_batch, ids_batch) in embedding_reader(batch_size=batch_size):
+            consecutive_ids = ids_batch["i"].to_numpy()
+            trained_index.add_with_ids(vec_batch, consecutive_ids)
+            if embedding_ids_df_handler:
+                all_ids.append(ids_batch)
+
+        if embedding_ids_df_handler:
+            embedding_ids_df_handler(pd.concat(all_ids), 0)
+    
+        if make_direct_map:
+            trained_index.make_direct_map()
+
+        return index_optimizer(trained_index, "")
+
+def _add_embeddings_to_index_from_hdfs_distributed(
+    trained_index_path: str,
+    trained_index_key: str,
+    embedding_root_dir: str,
+    embedding_count: int,
```

Comment:
`embedding_count` is not used. I guess it is because of `starmap` and the unpacked argumetns of `TrainedIndex`. 
I wonder if we can move the following code
```
small_indexes = []
big_indexes = []
for trained_index in trained_indexes:
    if trained_index.embedding_count < big_index_threshold:
        small_indexes.append(trained_index)
    else:
        big_indexes.append(trained_index)
```
outside of add_embeddings_to_indexes_from_hdfs and pass it two arguments `small_indexes` and `big_indexes` instead, so that we can get rid of a unused argument (`embedding_count`) here.
WDYT ?